miner/worker.go

```diff
@@ -893,10 +900,32 @@ func (w *worker) commitTransactions(env *environment, txs *transactionsByPriceAn
 	}
 	var coalescedLogs []*types.Log
 
+	//append the tx DAG transaction to the block
+	appendTxDAG := func() {
```

Comment:
I think this method is better called in `generateWork`, It may be called multiple times in `commitTransactions`.

Reply:
Indeed.